tests/components/alert/test_init.py

```diff
@@ -317,10 +320,286 @@ async def test_skipfirst(hass: HomeAssistant, mock_notifier: list[ServiceCall])
+    assert last_event.data[notify.ATTR_TITLE] == TEST_ENTITY
+
+    hass.states.async_set(TEST_ENTITY, STATE_OFF)
+    expected_notifications += 1
+    await hass.async_block_till_done()
+    assert len(mock_notifier) == expected_notifications
+    last_event = mock_notifier[-1]
+    assert last_event.data[notify.ATTR_MESSAGE] == TEST_ENTITY
+    assert last_event.data[notify.ATTR_TITLE] == TEST_ENTITY
+
+
+async def test_delete_and_reload(
+    hass: HomeAssistant, mock_notifier: list[ServiceCall], hass_admin_user: MockUser
+) -> None:
+    """Test reload after deleting alerts (for code coverage)."""
+    assert await async_setup_component(hass, DOMAIN, TEST_CONFIG)
+    expected_notifications = 0
+    hass.states.async_set(TEST_ENTITY, STATE_ON)
+    expected_notifications += 1
+    await hass.async_block_till_done()
+
+    assert hass.states.get(ENTITY_ID).state == STATE_ON
+    assert len(hass.states.async_entity_ids()) == 2
+    assert len(mock_notifier) == expected_notifications
+
+    state_1 = hass.states.get(ENTITY_ID)
+    assert state_1 is not None
+    assert state_1.name == NAME
+
+    # Delete the alert from yaml, leaving only the domain, with no alerts
+    config = deepcopy(TEST_CONFIG)
+    del config[DOMAIN][NAME]
+
+    with patch(
+        "homeassistant.config.load_yaml_config_file",
+        autospec=True,
+        return_value=config,
+    ):
+        await hass.services.async_call(
+            DOMAIN,
+            SERVICE_RELOAD,
+            blocking=True,
+            context=Context(user_id=hass_admin_user.id),
+        )
+        await hass.async_block_till_done()
+
+    assert len(mock_notifier) == expected_notifications
+    assert len(hass.states.async_entity_ids()) == 1
+    assert TEST_ENTITY in hass.states.async_entity_ids()
+
+    state_1 = hass.states.get(ENTITY_ID)
+    assert state_1 is None
+
+    # Delete the domain completely, and reload again
+    config = deepcopy(TEST_CONFIG)
+    del config[DOMAIN]
+    with patch(
+        "homeassistant.config.load_yaml_config_file",
+        autospec=True,
+        return_value=config,
+    ):
+        await hass.services.async_call(
+            DOMAIN,
+            SERVICE_RELOAD,
+            blocking=True,
+            context=Context(user_id=hass_admin_user.id),
+        )
+        await hass.async_block_till_done()
+
+    assert len(mock_notifier) == expected_notifications
+    assert len(hass.states.async_entity_ids()) == 1
+    assert TEST_ENTITY in hass.states.async_entity_ids()
+
+    state_1 = hass.states.get(ENTITY_ID)
+    assert state_1 is None
+
+
+async def test_from_storage(hass: HomeAssistant) -> None:
+    """Test the from_storage method of the alert."""
+    config = deepcopy(TEST_CONFIG[DOMAIN][NAME])
+    config[CONF_ID] = NAME
+    config[CONF_CAN_ACK] = False
+    config[CONF_REPEAT] = []
+    entity = alert.Alert.from_storage(config)
+    entity.entity_id = ENTITY_ID
+
+    assert entity._watched_entity_id == TEST_ENTITY
```

Comment:
Remove this

Reply:
See other discussion about `from_storage`, I can't remove this until I resolve that discussion.